articles/tutorials/qft-tutorial.md

```diff
@@ -0,0 +1,624 @@
+
+namespace Quantum.Operations
+{
+    class Driver
+    {
+        static void Main(string[] args)
+        {
+            using (var qsim = new QuantumSimulator())
+            {
+                Perform3QubitQFT.Run(qsim).Wait();
+            }
+            
+            System.Console.WriteLine("Press any key to continue...");
+            Console.ReadKey();
+        }
+    }
+}
+
+```
+Run the application, and your output should match that below.
+The program will exit after you press a key.
+***
+
+```Output
+Initial state |000>:
+# wave function for qubits with ids (least to most significant): 0;1;2
+|0>:	 1.000000 +  0.000000 i	 == 	******************** [ 1.000000 ]     --- [  0.00000 rad ]
+|1>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|2>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|3>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|4>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|5>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|6>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|7>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+After:
+# wave function for qubits with ids (least to most significant): 0;1;2
+|0>:	 0.353553 +  0.000000 i	 == 	***                  [ 0.125000 ]     --- [  0.00000 rad ]
+|1>:	 0.353553 +  0.000000 i	 == 	***                  [ 0.125000 ]     --- [  0.00000 rad ]
+|2>:	 0.353553 +  0.000000 i	 == 	***                  [ 0.125000 ]     --- [  0.00000 rad ]
+|3>:	 0.353553 +  0.000000 i	 == 	***                  [ 0.125000 ]     --- [  0.00000 rad ]
+|4>:	 0.353553 +  0.000000 i	 == 	***                  [ 0.125000 ]     --- [  0.00000 rad ]
+|5>:	 0.353553 +  0.000000 i	 == 	***                  [ 0.125000 ]     --- [  0.00000 rad ]
+|6>:	 0.353553 +  0.000000 i	 == 	***                  [ 0.125000 ]     --- [  0.00000 rad ]
+|7>:	 0.353553 +  0.000000 i	 == 	***                  [ 0.125000 ]     --- [  0.00000 rad ]
+```
+
+When called on the full-state simulator, `DumpMachine()` provides these mutliple representations of the quantum state's wavefunction. 
+The possible states of an $n$-qubit system can be represented by $2^n$ computational basis states, each with a corresponding complex coefficient (simply an amplitude and a phase).
+The computational basis states correspond to all the possible binary strings of length $n$---that is, all the possible combinations of qubit states $\ket{0}$ and $\ket{1}$, where each binary digit corresponds to an individual qubit.
+
+The first row provides a comment with the IDs of the corresponding qubits in their significant order.
+Qubit `2` being the "most significant" simply means that in the binary representation of basis state vector $\ket{i}$, the state of qubit `2` corresponds to the left-most digit. 
```

Comment:
Incidentally, I'd say "leftmost" and "rightmost," not "left-most" and "right-most"; but this repo currently uses the latter consistently.

Reply:
I was also unsure, but figured consistency is king! 🤷‍♂